pkg/apis/core/types.go

```diff
@@ -3505,6 +3529,12 @@ type EphemeralContainerCommon struct {
 	// +featureGate=InPlacePodVerticalScaling
 	// +optional
 	ResizePolicy []ContainerResizePolicy
+	// Restart policy for the container.
+	// This may only be set for init containers. You cannot set this field on
```

Comment:
nit: Any reason to call out as not supported for ephemeral containers when we explicitly say it is only for init containers?

Reply:
https://github.com/kubernetes/kubernetes/pull/116429#discussion_r1131800085

added the comment to just explicitly make sure that it should not be set inside of the EphemeralContainerCommon type.
(we added this field to EphemeralContainerCommon type as it is an alias of the Container type)